### PR TITLE
[DEV-618] BTC Output Node

### PR DESCRIPTION
**Pull request by davidleomay**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Review by TonySemikin — APPROVED** [review]

File: infrastructure/bicep/btc-node.bicep [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        version: 'latest'
+      }
+      osDisk: {
+        name: vmDiskName
+        createOption: 'FromImage'
+        caching: 'ReadWrite'
+        managedDisk: {
+          storageAccountType: 'StandardSSD_LRS'
+        }
+        diskSizeGB: 1023
```
Comment (Krysh90) [inline infrastructure/bicep/btc-node.bicep:117]: shouldn't this be a value of power of 2? Or is there a special reason to have it at 2 ^ 10 - 1?

File: infrastructure/bicep/dfx-api.bicep [hunk 2]
```diff
@@ -736,121 +692,19 @@ module nodes 'defi-node.bicep' = [for node in nodeProps: {
...
+  params: {
+    location: location
+    pipName: node.pipName
+    vmName: node.vmName
+    vmDiskName: node.vmDiskName
+    nsgName: node.nsgName
+    nicName: node.nicName
+    vmUser: btcVmUser
+    vmPassword: btcVmPassword
+    subnetId: virtualNet.properties.subnets[1].id
```
Comment (Krysh90) [inline infrastructure/bicep/dfx-api.bicep:706]: Just curious 
how many subnets do we have here?
Why not having everything within one subnet (index 0)?

**Review by Krysh90 — APPROVED** [review]

File: infrastructure/bicep/btc-node.bicep (hunk 1 above)
Comment (davidleomay) [reply infrastructure/bicep/btc-node.bicep:117]: image
Don't ask me why 🤔

File: infrastructure/bicep/dfx-api.bicep (hunk 2 above)
Comment (davidleomay) [reply infrastructure/bicep/dfx-api.bicep:706]: Good question ... I had to dig a little deeper 😉  I wanted the VM not to have access to other resources (DB, DeFiChain Nodes), therefore the second subnet.
But we could probably remove one NSG, as they are the same for both VMs.

**Review by Krysh90 — APPROVED** [review]

**Pull request merged**